.github/workflows/linux.yml

```diff
@@ -26,37 +14,44 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        build: [clang-6, gcc-7, gcc-8, gcc-9]
+        build: [clang-6-ssl, gcc-7-ssl, gcc-8-ssl, gcc-9-ssl, gcc-9]
+
         include:
-        - build: clang-6
-          os: ubuntu-latest
+        - build: clang-6-ssl
           INSTALL: clang-6.0
           C_COMPILER: clang-6.0
           CXX_COMPILER: clang++-6.0
+          extra_cmake_flags: -DWITH_OPENSSL=ON
 
-        - build: gcc-7
-          os: ubuntu-latest
-          INSTALL: gcc-7 g++-7
+        - build: gcc-7-ssl
+          INSTALL: gcc-7 g++-7 libssl-dev
           C_COMPILER: gcc-7
           CXX_COMPILER: g++-7
+          extra_cmake_flags: -DWITH_OPENSSL=ON
 
-        - build: gcc-8
-          os: ubuntu-latest
+        - build: gcc-8-ssl
           INSTALL: gcc-8 g++-8
           C_COMPILER: gcc-8
           CXX_COMPILER: g++-8
+          extra_cmake_flags: -DWITH_OPENSSL=ON
+
+        - build: gcc-9-ssl
+          INSTALL: gcc-9 g++-9
```

Comment:
ups, it look like i didn't intall libssl-dev in every scenario, but somehow it works...

Reply:
It could be already installed, but anyway let's make it explicit dependency

Reply:
I like your option much more :+1: Wasn't aware that you can make a cartesian product of matrixes this way